exporters/memory/include/opentelemetry/exporters/memory/in_memory_span_data.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/span_data.h"
+#include "src/common/atomic_unique_ptr.h"
+#include "src/common/circular_buffer.h"
+#include "src/common/circular_buffer_range.h"
+
+using opentelemetry::sdk::common::CircularBuffer;
+using opentelemetry::sdk::common::CircularBufferRange;
+using opentelemetry::sdk::common::AtomicUniquePtr;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+/**
+ * A wrapper class holding in memory exporter data
+ */
+class InMemorySpanData final
+{
+public:
+  InMemorySpanData();
+
+  void Add(std::unique_ptr<sdk::trace::SpanData> data) noexcept;
+
+  std::vector<sdk::trace::SpanData> GetSpans() noexcept;
```

Comment:
Make `Add` and the constructor private and add the `InMemorySpanExporter` as a friend class.

Reply:
I'm not sure how to configure the configure the BUILD file to enable this. This would require adding the in_memory_span_data and in_memory_span_exporter as dependencies of each other, and would create a circular dependency. Any suggestions on how this can be done?